### PR TITLE
Chore: Refactor AppLogs to TS

### PR DESCRIPTION
**Pull request by rique223**
## Proposed changes (including videos or screenshots)

Rewrote the AppLogs component to TS. Also, improved some error handling logic.

### CODE REVIEW

File: apps/meteor/client/views/admin/apps/AppLogs.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+type data = AppOverview & { logs: logs; logsError: string };
+
+const useAppWithLogs = ({ id }: { id: string }): data => {
+	const dispatchToastMessage = useToastMessageDispatch();
+	const t = useTranslation();
+
+	const [data, setData] = useSafely(useState<data>({} as data));
+	const [logsError, setLogsError] = useState('');
+	const getAppData = useEndpoint('GET', `/apps/${id}`) as any;
+	const getAppLogs = useEndpoint('GET', `/apps/${id}/logs`) as any;
```
[the remark this answers is not in the document]
Comment (rique223) [reply apps/meteor/client/views/admin/apps/AppLogs.tsx:40]: I am not really sure. I've spent a lot of time on this and wasn't able to come up with a way for it to stop breaking when I don't type it like this.

[the remark this answers is not in the document]
Comment (rique223) [reply apps/meteor/client/views/admin/apps/AppLogs.tsx]: Solved

File: packages/rest-typings/src/apps/index.ts [hunk 2]
```diff
@@ -5,16 +5,21 @@ import type { IPermission } from '@rocket.chat/apps-engine/definition/permission
 import type { ISetting } from '@rocket.chat/apps-engine/definition/settings';
 import type { IUIActionButton } from '@rocket.chat/apps-engine/definition/ui';
 import type { AppScreenshot, App, FeaturedAppsSection } from '@rocket.chat/core-typings';
+import type { ILogItem } from '@rocket.chat/core-typings/src/ILogs';
```
Comment (dougfabris) [inline packages/rest-typings/src/apps/index.ts:8]: You should add the `ILogs.ts` type file on the index to get the types from the root
```suggestion
import type { ILogItem } from '@rocket.chat/core-typings';
```

File: apps/meteor/client/views/admin/apps/LogItem.tsx [hunk 3]
```diff
@@ -1,16 +1,12 @@
+import { ILogEntry } from '@rocket.chat/core-typings/src/ILogs';
```
Comment (dougfabris) [inline apps/meteor/client/views/admin/apps/LogItem.tsx:1]:
```suggestion
import { ILogEntry } from '@rocket.chat/core-typings';
```

File: apps/meteor/client/views/admin/apps/AppLogs.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import { useEndpointData } from '../../../hooks/useEndpointData';
+import { useFormatDateAndTime } from '../../../hooks/useFormatDateAndTime';
+import { AsyncStatePhase } from '../../../lib/asyncState';
+import AccordionLoading from './AccordionLoading';
+import LogItem from './LogItem';
+
+const AppLogs = ({ id }: { id: string }): ReactElement => {
+	const t = useTranslation();
+	const formatDateAndTime = useFormatDateAndTime();
+	const { value, phase: logsPhase, error } = useEndpointData(`/apps/${id}/logs`);
```
Comment (dougfabris) [inline apps/meteor/client/views/admin/apps/AppLogs.tsx:14]: Why not `react-query`??

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json [hunk 5]
```diff
@@ -4858,6 +4859,7 @@
   "Unignore": "Unignore",
   "Uninstall": "Uninstall",
   "Unit_removed": "Unit Removed",
+  "Unknown_error": "An Unknown Error occured",
```
Comment (dougfabris) [inline apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json:4862]:
```suggestion

```

**Review by dougfabris — CHANGES_REQUESTED** [review]

**Review by juliajforesti — APPROVED** [review]

**Review by dougfabris — APPROVED** [review]

**Pull request merged**